src/Microsoft.AspNet.Razor.Runtime/TagHelpers/HtmlElementNameAttribute.cs

```diff
@@ -29,22 +31,37 @@ public HtmlElementNameAttribute([NotNull] string tag)
         /// <param name="additionalTags">Additional HTML tag names for the <see cref="TagHelper"/> to target.</param>
         public HtmlElementNameAttribute([NotNull] string tag, [NotNull] params string[] additionalTags)
         {
-            if (additionalTags.Contains(null))
-            {
-                throw new ArgumentNullException(
-                    nameof(additionalTags),
-                    Resources.FormatHtmlElementNameAttribute_AdditionalTagsCannotContainNull(nameof(additionalTags)));
-            };
-
             var allTags = new List<string>(additionalTags);
             allTags.Add(tag);
 
+            foreach (var tagName in allTags)
+            {
+                ValidateTagName(tagName);
+            }
+
             Tags = allTags;
         }
 
         /// <summary>
         /// An <see cref="IEnumerable{string}"/> of tag names for the <see cref="TagHelper"/> to target.
         /// </summary>
-        public IEnumerable<string> Tags { get; private set; }
+        public IEnumerable<string> Tags { get; }
+
+        private static void ValidateTagName(string tagName)
+        {
+            if (string.IsNullOrWhiteSpace(tagName))
+            {
+                throw new ArgumentException(
+                    Resources.HtmlElementNameAttribute_ElementNameCannotBeNullOrWhitespace,
+                    nameof(tagName));
```

Comment:
do not `throw` an `ArgumentException` using a parameter name of a private method. the parameter that's invalid is either `tag` or `additionalTags`.  pass the correct name into this method.